src/v/serde/parquet/tests/column_writer_bench.cc

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "bytes/random.h"
+#include "container/fragmented_vector.h"
+#include "random/generators.h"
+#include "serde/parquet/column_writer.h"
+#include "serde/parquet/schema.h"
+#include "serde/parquet/value.h"
+
+#include <seastar/testing/perf_tests.hh>
+
+#include <cstdint>
+
+using namespace serde::parquet;
+
+namespace {
+
+schema_element leaf_node(
+  ss::sstring name,
+  field_repetition_type rep_type,
+  physical_type ptype,
+  logical_type ltype = logical_type{}) {
+    return {
+      .type = ptype,
+      .repetition_type = rep_type,
+      .path = {std::move(name)},
+      .logical_type = ltype,
+    };
+}
+
+chunked_vector<iobuf> gen_rand_iobufs(size_t num_bufs, size_t buf_len) {
+    chunked_vector<iobuf> bufs;
+    bufs.reserve(num_bufs);
+    for (size_t i = 0; i < num_bufs; i++) {
+        bufs.push_back(random_generators::make_iobuf(buf_len));
+        // Share ahead of time to allocate underlying deleters.
+        bufs[i].share(0, bufs[i].size_bytes());
+    }
+
+    return bufs;
+}
+
+template<typename T>
+chunked_vector<T> gen_rand(size_t num, std::function<T()> rand_fn) {
+    chunked_vector<T> ret;
+    ret.reserve(num);
+    for (size_t i = 0; i < num; i++) {
+        ret.push_back(rand_fn());
+    }
+    return ret;
+}
+
+constexpr int64_t page_size = 512 * 1024;
+constexpr size_t num_rows = 1'000'000;
+constexpr size_t num_values = 1'000;
+
+} // namespace
+
+class column_writer_bench {
+public:
+    ss::future<size_t> bench(
+      const schema_element& element, std::function<value(size_t)> value_fn) {
+        auto writer = column_writer(
+          element,
+          {
+            .compress = true,
+          });
+
+        perf_tests::start_measuring_time();
+        for (size_t i = 0; i < num_rows; i++) {
+            writer.add(value_fn(i), rep_level{0}, def_level{0});
```

Comment:
Not sure if it's a good idea to have `value_fn` inside the measured region like this. How does it change if you just replace this with a local variable (passed through `perf_tests::do_not_optimize`)?

Reply:
Moved it outside of the measured region.